### PR TITLE
Verify that policyfiles have a valid revision id field

### PR DESCRIPTION
**Pull request by danielsdeleo**
#### :construction: Pending Erchef Validation (400) Fixes ####

According to the draft RFC, Policyfiles are required to have a Revision ID field. https://github.com/chef/chef-rfc/pull/91

Add tests that ensure a server validates the revision ID. Also updates 400 message body expectations to match erchef's validation style.

Related PRs:

* erchef: https://github.com/chef/oc_erchef/pull/98
* Chef Zero: https://github.com/chef/chef-zero/pull/116

### CODE REVIEW

File: spec/api/policies/complete_endpoint_spec.rb [hunk 1]
```diff
@@ -209,13 +206,68 @@ def mutate_json(data)
...
+
+            let(:expected_error_message) { "Field 'revision_id' invalid" }
+
+            include_examples "an invalid policy document"
+
+          end
+
+          context "because revision id field is larger than 255 characters" do
+
+            let(:long_revision_id_is_long) { "f" * 256 }
```
Comment (marcparadise) [inline spec/api/policies/complete_endpoint_spec.rb]: Could you add an at-the-limit 255 test?

**Comment by marcparadise** [comment]
:+1:

**Comment by danielsdeleo** [comment]
@marcparadise Added edge cases tests. Now verify that all acceptable characters are accepted, and that maximum length fields are accepted for name, revision id, and identifiers in cookbook_locks
